### PR TITLE
Feature/cqi 157: add security headers

### PR DESCRIPTION
**Pull request by lruzicki**
Wait until this PR is merged:
https://github.com/openimis/openimis-be-core_py/pull/301

Ticket:
https://openimis.atlassian.net/browse/CQI-157

Changes:
- headers recommended by OWASP were added to settings.py
- some of them cannot be set diretcly in settings.py and proper middleware will be created to enchance security

How was it tested?
- headers were visible on PROD mode
- redirection from HTTP to HTTPS with header works as intended

### CODE REVIEW

File: openIMIS/openIMIS/settings.py [hunk 1]
```diff
@@ -225,8 +224,11 @@ def SITE_URL():
     "django.middleware.common.CommonMiddleware",
     "django.middleware.csrf.CsrfViewMiddleware",
     "django.contrib.auth.middleware.AuthenticationMiddleware",
+    "core.middleware.SecurityHeadersMiddleware",
 ]
 
+MODE = os.environ.get("MODE", "DEV")
```
Comment (dborowiecki) [inline openIMIS/openIMIS/settings.py]: I don't think we can force this decision by us. Please check it with @delcroip if we can set DEV mode as default or revert this change.

File: openIMIS/openIMIS/settings.py [hunk 2]
```diff
@@ -297,6 +299,24 @@ def SITE_URL():
...
+    CSRF_COOKIE_HTTPONLY = True
+    CSRF_COOKIE_SAMESITE = 'Lax'
+
+    SECURE_BROWSER_XSS_FILTER = True
+    SECURE_CONTENT_TYPE_NOSNIFF = True
+    SECURE_HSTS_SECONDS = 63072000
+    SECURE_HSTS_INCLUDE_SUBDOMAINS = True
+    SECURE_HSTS_PRELOAD = True
+    SECURE_SSL_REDIRECT = True
+
```
Comment (dborowiecki) [inline openIMIS/openIMIS/settings.py]: In this piece of code the comment says `# Enhance security in production`. However, this parameters are set only on DEV mode. Can you make adjustments to have consistency?

**Review by dborowiecki — CHANGES_REQUESTED** [review]
Please find my comments.

File: openIMIS/openIMIS/settings.py (hunk 2 above)
Comment (lruzicki) [reply openIMIS/openIMIS/settings.py]: You are right it should be only for PROD

File: openIMIS/openIMIS/settings.py (hunk 1 above)
Comment (lruzicki) [reply openIMIS/openIMIS/settings.py]: We can keep just MODE without setting up default one.

**Review by dborowiecki — CHANGES_REQUESTED** [review]
@lruzicki I think it's almost ready but I'd like to have short README section explainig the setup process of CSRF on dev and prod (see https://docs.djangoproject.com/en/5.0/ref/settings/#csrf-trusted-origins for reference)

**Comment by lruzicki** [comment]
> I think it's almost ready but I'd like to have short README section explainig the setup process of CSRF on dev and prod (see https://docs.djangoproject.com/en/5.0/ref/settings/#csrf-trusted-origins for reference)

@dborowiecki I've added more info, not only the env variable description but also a short section.

**Review by dborowiecki — APPROVED** [review]
